### PR TITLE
fix: Remove unknown data

### LINKED ISSUE #339: Unknown in קיום נסיעות component
Screenshot from 2023-12-25 09-28-12

### PR DESCRIPTION
**Pull request by IdanDover**
# Description
fixed #339 problem on frontend side where there were "unknown" bus provider
need to check the agency list and add a new agency to be like the DB

## screenshots
image
image

### CODE REVIEW

**Comment by NoamGaash** [comment]
https://open-bus-stride-api.hasadna.org.il/gtfs_agencies/list

I don't see any agency with that ID in the gtfs data
interesting

**Comment by IdanDover** [comment]
If you will log the data recieved at  the useGroupby hook you will see there is a alot of operator_ref with the value of 22. Maybe it's some old data in the postgress that was not updated or deleted in some way

**Comment by NoamGaash** [comment]
It worth an exploration using the [Swagger](https://open-bus-stride-api.hasadna.org.il/docs#/siri/list__siri_rides_list_get).

image

I've tried to run this query, and seems like it takes forever
https://open-bus-stride-api.hasadna.org.il/siri_rides/list?get_count=false&gtfs_route__date_from=2024-01-20&gtfs_route__date_to=2024-01-27&gtfs_route__operator_refs=22&order_by=id%20asc

**Comment by NoamGaash** [comment]
ok it worked - seems like we get data from the TLV light rail (Dankal)!
How exciting

File: src/api/groupByService.ts [hunk 1]
```diff
@@ -116,12 +116,15 @@ export function useGroupBy({
...
-            ),
-          })),
+          data
+            .map((dataRecord) => ({
+              ...dataRecord,
+              operator_ref: agencyList.find(
+                (agency) => agency.agency_id === String(dataRecord.operator_ref),
+              ),
+            }))
+            //NOTE: the problem comes from agencyList number 22 is not found. need to add the agency or remove it from the DB
```
Comment (NoamGaash) [inline src/api/groupByService.ts:126]:
```suggestion
            // should filter operator 22 (which is the Dankal TLV light train)
```

**Review by NoamGaash — APPROVED** [review]
Thanks for your investigation :)
that's so cool.
let's fix the comment and merge that

**Comment by IdanDover** [comment]
no problem, I changed the comment to what you specified. happy to contribute :)

**Pull request merged**